crates/matrix-sdk/src/oidc/registrations.rs

```diff
@@ -20,22 +28,34 @@ use std::{
     path::PathBuf,
 };
 
-use matrix_sdk::oidc::types::registration::{
+use mas_oidc_client::types::registration::{
     ClientMetadata, ClientMetadataVerificationError, VerifiedClientMetadata,
 };
 use serde::{Deserialize, Serialize};
 use url::Url;
 
+/// Errors related to persisting OIDC registrations.
 #[derive(Debug, thiserror::Error)]
 pub enum OidcRegistrationsError {
+    /// The supplied base path is invalid.
     #[error("Failed to use the supplied base path.")]
     InvalidBasePath,
+    /// An error occurred whilst loading the registration data.
     #[error("Failed to load registrations file: {message}")]
-    LoadFailure { message: String },
+    LoadFailure {
+        /// The error message.
+        message: String,
+    },
+    /// The stored metadata doesn't match the supplied metadata: any stored
+    /// registrations are no longer relevant for this client.
     #[error("Metadata mismatch, ignoring any stored registrations.")]
     MetadataMismatch,
```

Comment:
So these two errors are publicaly exposed, but they can't be caught if calling public functions. They're only used to signify internal errors, that we immediately catch and ignore.

Instead of having these two variants, could we, for each error use location, warn immediately with the error message, and early return an empty `Err(())`? Then we wouldn't have these two error variants in the public API. Lean APIs are nice.

Reply:
Ok yep, that looks much better to me 👍